packages/engine/src/datastore/table/pool/mod.rs

```diff
@@ -69,7 +69,7 @@ pub trait BatchPool<B: Batch>: Send + Sync {
     ///
     /// - If `index` is out of bounds, or
     /// - if the `Batch` is currently borrowed as a [`BatchReadProxy`] or [`BatchWriteProxy`].
-    fn remove(&mut self, index: usize) -> String;
+    fn remove(&mut self, index: usize) -> B;
```

Comment:
It's too far away from the changed lines for me to suggest changes but can you update the doc-comments on these methods. The description of the return values is incorrect.